Production `next build` fails: TAG_COLOR missing Signal/Builder keys (20-18 additive-tag FLOOR miss)

## Problem

`pnpm --filter @stave/app build` (production `next build`, which runs full `tsc`) fails:

```
src/components/IRInspectorPanel.tsx(89,7): error TS2739: Type '{ ... }' is missing
the following properties from type 'Record<PatternIR["tag"], string>': Signal, Builder
```

Phase 20-18 added `Signal` and `Builder` to the `PatternIR["tag"]` union (Layer-1/2 IR substrate), but the app-side `TAG_COLOR: Record<PatternIR["tag"], string>` map in `IRInspectorPanel.tsx:89` was never given color tokens for them. The map is no longer exhaustive over the tag union.

## Why it was latent

`next build` is the ONLY gate that runs full `tsc` over the app. The vitest suites (transform-only, no full typecheck) stayed green, so this has been broken since 20-18 (2026-05-20) and only surfaced now while preparing the first production deploy.

This is a **PV54 additive-tag FLOOR-grep obligation** miss — 20-18 updated the editor-package FLOOR switches but this consumer lives in `packages/app`, outside that audit's scope.

## Fix

Add `Signal` and `Builder` entries to `TAG_COLOR` with distinct design-system color tokens (`var(--ir-signal, …)` / `var(--ir-builder, …)`), matching the existing placeholder-fallback convention (see `Track`).

## Surfaced by

First production-deploy prep (Vercel). Build must typecheck clean before deploy.

--- a/packages/app/src/components/IRInspectorPanel.tsx
+++ b/packages/app/src/components/IRInspectorPanel.tsx
@@ -135,6 +135,14 @@ const TAG_COLOR: Record<PatternIR["tag"], string> = {
   // pink (#ec4899) or Param's orange (#fb923c). 20-12 design-system pass
   // lands the final swatch token.
   Track:    "var(--ir-track, #94a3b8)",
+  // Phase 20-18 — Signal/Builder Layer-1/2 IR substrate tags. Signal =
+  // continuous signals (sine/perlin/irand/…); Builder = chord/arrange/
+  // cat/transpose construction roots. Distinct placeholder hues (indigo-400
+  // / green-400) that don't collide with neighbours (Cycle #06b6d4,
+  // Play #10b981); the final swatch tokens (--ir-signal / --ir-builder)
+  // land in the design-system pass, same as Track's placeholder.
+  Signal:   "var(--ir-signal, #818cf8)",
+  Builder:  "var(--ir-builder, #4ade80)",
 };
 
 // summarize / children moved to IRInspectorChrome.ts (Phase 20-04 wave δ)

--- a/packages/app/vercel.json
+++ b/packages/app/vercel.json
@@ -1,0 +1,4 @@
+{
+  "$schema": "https://openapi.vercel.sh/vercel.json",
+  "buildCommand": "cd ../.. && pnpm turbo run build --filter=@stave/app"
+}

